lib/network/webserver.js

```diff
@@ -0,0 +1,352 @@
+'use strict'; // eslint-disable-line
+
+const http = require('http');
+const https = require('https');
+const assert = require('assert');
+const Logger = require('werelogs').Logger;
+const dhparam = require('../https/dh2048').dhparam;
+const ciphers = require('../https/ciphers').ciphers;
+const errors = require('../errors');
+
+class WebServer {
+
+    /**
+     * @constructor
+     *
+     * @param {number} port - Port to listen into
+     * @param {werelogs.Logger} logger - Logger object
+     */
+    constructor(port, logger) {
+        assert.strictEqual(typeof port, 'number', 'Port must be a number');
+        assert.strictEqual(logger instanceof Logger, true,
+            'Logger must be an instance of werelogs');
+        this._keepAlive = false;
+        this._noDelay = true;
+        this._cbOnListening = null;
```

Comment:
Can you set theses callback to a dummy function ```() => {}```, so you can avoid to test their existence each time ?